File: 19_07_2024_Lucky_Numbers_in_a_Matrix.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<Integer> luckyNumbers(int[][] matrix) {
        int rows = matrix.length;
        int cols = matrix[0].length;
        List<Integer> ans = new ArrayList<>();

        for (int r_idx = 0; r_idx < rows; r_idx++) {
            int minEle = matrix[r_idx][0];
            int col_idx = 0;

            // Find the minimum element in the row and its column index
            for (int c = 1; c < cols; c++) {
                if (matrix[r_idx][c] < minEle) {
                    minEle = matrix[r_idx][c];
                    col_idx = c;
                }
            }

            // Check if this minimum element is the maximum in its column
            int r;
            for (r = 0; r < rows; r++) {
                if (matrix[r_idx][col_idx] < matrix[r][col_idx]) {
                    break;
                }
            }

            // If the loop completes, it means minEle is a lucky number
            if (r == rows) {
                ans.add(minEle);
            }
        }

        return ans;
    }
}

File: 19_07_2024_Lucky_Numbers_in_a_Matrix.py
class Solution(object):
    def luckyNumbers (self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[int]
        """
        rows=len(matrix)
        ans=0
        for r_idx,row in enumerate(matrix):
            min_ele=min(row)
            col_idx=row.index(min_ele)
            r=0
            while r<rows:
                if matrix[r_idx][col_idx]<matrix[r][col_idx]:
                    break
                r+=1
            if r==rows:
                return [min_ele]
